independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/name/InvalidNamedInjectionPointTest.java

```diff
@@ -0,0 +1,43 @@
+package io.quarkus.arc.test.name;
+
+import static org.junit.jupiter.api.Assertions.assertInstanceOf;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import javax.enterprise.context.Dependent;
+import javax.enterprise.inject.spi.DefinitionException;
+import javax.inject.Inject;
+import javax.inject.Named;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.arc.test.ArcTestContainer;
+
+public class InvalidNamedInjectionPointTest {
+
+    @RegisterExtension
+    public ArcTestContainer container = ArcTestContainer.builder()
+            .beanClasses(Foo.class, Consumer.class)
+            .shouldFail()
+            .build();
+
+    @Test
+    public void test() {
+        assertNotNull(container.getFailure());
+        assertInstanceOf(DefinitionException.class, container.getFailure());
+        assertTrue(container.getFailure().getMessage().contains("@Named without value may not be used on method parameter"));
+    }
+
+    @Named("foo")
+    @Dependent
+    static class Foo {
+    }
+
+    @Dependent
+    static class Consumer {
+        @Inject
+        Consumer(@Named Foo foo) {
```

Comment:
In theory, we could be a bit more lenient and if a param name is available then just use it instead of throwing a definition exception. On the other hand, `@Named` at injection points should not be rarely used. So it's probably not worth the effort...

Reply:
True. The spec says it's a definition error, and while we could improve on this, I didn't feel like spending more time on this particular corner case :-)

Reply:
:+1: when this rule was created it was not possible to store the param names in the class file ;-)